openedx/features/partners/tests/test_login.py

```diff
@@ -0,0 +1,113 @@
+from student.tests.factories import UserFactory
+
+
+class ResetPasswordTestCases(APITestCase):
+
+    def setUp(self):
+        self.partner_login_end_point = reverse('partner_login', args=['give2asia'])
+        self.user = UserFactory(username='testuser', password='Abc12345',email='abc@test.com')
+
+    def test_login_with_correct_credentials(self):
+        """
+                Testing login with valid credentials
+        """
+        valid_data = {
+            'email': 'abc@test.com',
+            'password': 'Abc12345'
+        }
+
+        partner_1 = PartnerFactory(slug='give2asia')
+
+        PartnerUserFactory(user_id=self.user.id, partner_id=partner_1.id)
+
+        response = self.client.post(
+            self.partner_login_end_point,
+            data=valid_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_login_with_incorrect_password(self):
+        """
+                Providing incorrect password
+        """
+        invalid_data = {
+            'email': 'abc@test.com',
+            'password': 'incorrectpassword'
+        }
+        partner_1 = PartnerFactory(slug='give2asia')
+
+        PartnerUserFactory(user_id=self.user.id, partner_id=partner_1.id)
+
+        response = self.client.post(
+            self.partner_login_end_point,
+            data=invalid_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_login_with_not_registered_username(self):
+        """
+                Providing username which isn't registered yet
+        """
+        invalid_data = {
+            'email': 'efg@test.com',
+            'password': 'qwertyQ123'
+        }
+        partner_1 = PartnerFactory(slug='give2asia')
+        response = self.client.post(
+            self.partner_login_end_point,
+            data=invalid_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_login_with_username_not_affiliated_with_partner(self):
+        """
+                Providing username which isn't affiliated to any partner
+        """
+        invalid_data = {
+            'email': 'abc@test.com',
+            'password': 'Abc12345'
+        }
+        partner_1 = PartnerFactory(slug='give2asia')
+        response = self.client.post(
+            self.partner_login_end_point,
+            data=invalid_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_login_with_username_with_nonexistant_partner(self):
+        """
+                Not even a single partner exists
+        """
+        invalid_data = {
+            'email': 'abc@test.com',
+            'password': 'Abc12345'
+        }
+        response = self.client.post(
+            self.partner_login_end_point,
+            data=invalid_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
```

Comment:
This test is not what it states. Endpoint will throw error not because no partner existed but because your are trying to login specifically at `.../login/partners/give2asia` and it does not exists

Reply:
Zahid bhai this is what the test is for. To check when give2asia does not exist.

Reply:
I see Ive made a mistake with the naming and docstring. I'll correct it.

Reply:
In that case you can update doc `Not even a single partner exists`, because from this doc it seems you are explicitly testing the case as if no parter exists but you are not. You are only asserting that partner `give2asia` dose't exists and its url is not accessible.

That is why in this PR, i have flagged the fact that, method doc should cover whole test scenario and what data used for testing

Reply:
Corrected this issue.